src/array_api_typing/_namespace.py

```diff
@@ -0,0 +1,27 @@
+"""Static typing support for the array API standard."""
+
+__all__ = ["HasNamespace"]
+
+from typing import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class HasNamespace(Protocol):
+    """Protocol for classes that have an `__array_namespace__` method.
+
+    This is a runtime checkable protocol.
+
+    Example:
+    >>> import array_api_typing as xpt
+    >>>
+    >>> class MyArray:
+    ...     def __array_namespace__(self):
+    ...         return "my namespace"
+    >>>
+    >>> x = MyArray()
+    >>> print(isinstance(x, xpt.HasNamespace))
+    True
+
+    """
+
+    def __array_namespace__(self) -> object: ...  # noqa: PLW3201
```

Comment:
How about we use a generic return type here, with e.g. `types.ModuleType` (PEP 696) as the PEP 696 default?

Reply:
E.g. wrapper types

```
class Wrapper[T: HasNamespace]:
    wrapped: T
    def __array_namespace__(self):
        ns = wrapped. __array_namespace__()
        return WrapperNamespace(ns)  # <- attribute access & modifies ns
```

Reply:
> I'm not sure we want to limit to module types. 

Yea I agree that we probably shouldn't. I'm suggesting that we *default* to it. 

> Or `Any` ?

`Any` or `object` would also be fine as default, AFAIK.

Reply:
Done.